src/track/beats.h

```diff
@@ -1,44 +1,78 @@
-#ifndef BEATS_H
-#define BEATS_H
+#pragma once
 
-#include <QObject>
-#include <QString>
-#include <QList>
 #include <QByteArray>
+#include <QList>
+#include <QObject>
 #include <QSharedPointer>
+#include <QString>
 
+#include "proto/beats.pb.h"
+#include "util/logger.h"
 #include "util/memory.h"
+#include "util/types.h"
 
-namespace {
-    double kMaxBpm = 500;
-}
-
+namespace mixxx {
 class Beats;
-typedef QSharedPointer<Beats> BeatsPointer;
+using BeatsPointer = std::shared_ptr<Beats>;
+using BeatList = QList<track::io::Beat>;
+using SamplePos = double;
+using FrameNum = double;
+} // namespace mixxx
 
-class BeatIterator {
-  public:
-    virtual ~BeatIterator() {}
-    virtual bool hasNext() const = 0;
-    virtual double next() = 0;
-};
+#define BEAT_MAP_VERSION "BeatMap-1.0"
+// TODO(JVC) This two are deprecated
+#define BEAT_GRID_1_VERSION "BeatGrid-1.0"
+#define BEAT_GRID_2_VERSION "BeatGrid-2.0"
+
+#include "track/beatiterator.h"
+#include "track/timesignature.h"
+#include "track/track.h"
 
+namespace {
```

Comment:
An anonymous namespace in a header file doesn't work as intended

Reply:
Removed.
Still not clear why this constant about maximum BPM is there. This information must come from preference?
I leave a TODO for further investigation.